sympy/matrices/tests/test_matrices.py

```diff
@@ -2328,19 +2328,47 @@ def test_col_join():
 
 def test_row_insert():
     r4 = Matrix([[4, 4, 4]])
-    for i in range(-4, 5):
+    for i in range(-3, 3):
```

Comment:
Why is it needed to change prior tests? Were they incorrect?

Reply:
@moorepants  Yes, they were incorrect. Since earlier, even if the index went out of bound (like in the above case, `-4`, `4`, and `5` were out of bounds) or was a non integer, it didn't result in an error (which it should've). But that's been corrected by this PR.

Reply:
@sidhantnagpal  Sorry, but I couldn't understand what you are trying to say here. Could you be a bit more clear please?

Reply:
@sidhantnagpal  I have corrected the floats part (which @asmeurer  had asked) but I still don't understand why is the range (-4, 5) valid? I think the entire purpose of using this statement :
```
        if pos < 0:
            pos = self.rows + pos
```
was 
1. to make the negative indices `>=0`
and 
2. `pos` when exceeds `self.rows`, doesn't make sense to insert another row in place of a row which doesn't exist. This is why I think `-4`, `4`, and `5` are out of bounds.
Am I wrong here?
What do you think?

Reply:
@sidhantnagpal  ping